featuretools/entityset/entityset.py

```diff
@@ -620,19 +620,15 @@ def add_dataframe(self,
                 warnings.warn('Performing type inference on Dask or Koalas DataFrames may be computationally intensive. '
                               'Specify logical types for each column to speed up EntitySet initialization.')
 
+            created_index, index, dataframe = _create_index(index, make_index, dataframe)
```

Comment:
can we change the name of `_create_index`? Or maybe split it up into separate `_validate_index` and `_create_index` functions? (I see that this came from the `Entity` class, but in this context the name makes the three variables returned from it kind of confusing)

Reply:
Updating the function name to `_get_or_create_index` and changing `created_index` to a boolean `index_was_created` to better indicate whether the index column was created by Featuretools or not.